tests/e2e/helm.go

```diff
@@ -292,3 +292,68 @@ func lookUpInstalledHelmReleaseByName(
 
 	return nil, false, nil
 }
+
+// uninstallHelmChart uninstall Helm chart from the specified kubectl context and
+// namespace using the specified helm release name. When purge is true, it removes records
+// and make that name free to be reused for another installation.
+// Extra arguments can be any of the helm CLI uninstall flag arguments.
+func uninstallHelmChart(
+	kubectlOptions k8s.KubectlOptions,
+	helmReleaseName string,
+	purge bool,
+	extraArgs ...string,
+) error {
+	By(
+		fmt.Sprintf(
+			"uninstalling Helm chart by name %s",
+			helmReleaseName,
+		),
+	)
+
+	fixedArguments := []string{
+		"--debug",
+	}
+
+	extraArgs = append(extraArgs, fixedArguments...)
+
+	return helm.DeleteE(
+		GinkgoT(),
+		&helm.Options{
+			KubectlOptions: &kubectlOptions,
+			ExtraArgs: map[string][]string{
+				"delete": extraArgs,
+			},
+		},
+		helmReleaseName,
+		purge,
+	)
+}
+
+// uninstallHelmChartIfExists checks whether the specified named Helm release exists in
+// the provided kubectl context and namespace, logs it if it does and returns or
+// alternatively uninstall the Helm chart from the specified kubectl context and
+// namespace using the release name, extra arguments can be any of the helm
+// CLI install flag arguments.
+func uninstallHelmChartIfExists(
```

Comment:
**Recommended**: I think it would make more sense to transform the `uninstallHelmChart[IfExists]` into a member function of the Helm descriptor as well, that case we wouldn't need to query the namespace where the Helm chart is installed from the descriptor, we wouldn't need to pass the release name because it would be available from the descriptor.

IMO that fits the current organization better, but I do reckon it's on the edge - first I didn't comment this, only came back when I saw the uninstall test case.

What do you think @panyuenlau @Kuvesz @mihaialexandrescu ?

Reply:
I think it is a good idea from my side

Reply:
You can go ahead and implement it that way then, others can provide their thoughts later and we can change it if anything comes up.
But as you and me were the ones who thought up this part of the interface if we agree on it then it may not be such a dumb idea. 😄

Reply:
Fixed in [63ed2cd](https://github.com/banzaicloud/koperator/pull/987/commits/63ed2cdca59f233e329f912c042c39a21ecf3a7c)